Conditionally disable complaints per 1000

flacoman91 opened the pull request:
When any filters are applied other than Date,
- [x] Switch Map Shading dropdown to 2 button treatment on Landing Page
- [x] Disable the Complaints Per 1000 Button
- [x] Go back to raw Complaints Mode
- [x] Display warning message

![filter-apply](https://user-images.githubusercontent.com/1038306/77673972-44806f00-6f61-11ea-8213-fb8edac9781b.gif)

sephcoster commented on src/PerCapita.less:
```diff
@@ -1,7 +1,10 @@
+  button {
+    color: @black;
+    &.deselected {
+      background-color: @pacific-40;
```
It looks like this color may be slightly off compared to the date range selector:
![Screen Shot 2020-03-26 at 1 09 50 PM](https://user-images.githubusercontent.com/1028659/77675273-1c920b00-6f63-11ea-9834-aa0a109de00d.png)

The contrast difference with black text is subtle but makes it difficult to read.

Minor fix if you can get to it in this PR. Thanks!

flacoman91 replied:
changed.

JeffreyMFarley commented on src/__tests__/utils.spec.jsx:
```diff
@@ -149,5 +150,27 @@ describe('module::utils', () => {
+    it( 'handles some filters', () => {
+      const query = {
+        date: {},
+        bogus: {},
```
:+1:

flacoman91 commented:
Ok, updated colors, and added the narrative and fixed the tags filter.
![filter-apply-tags](https://user-images.githubusercontent.com/1038306/77682109-c4f89d00-6f6c-11ea-8243-ca342839a287.gif)
![filter-apply-2](https://user-images.githubusercontent.com/1038306/77682125-c924ba80-6f6c-11ea-90e9-ac9474136d52.gif)

the filters that trigger this are:
```
  'company',
  'company_public_response',
  'company_response',
  'consumer_consent_provided',
  'consumer_disputed',
  'issue',
  'product',
  'state',
  'submitted_via',
  'tags',
  'timely',
  'zip_code',

'has_narrative'
```

still up in the air about
```
  'company_received_max',
  'company_received_min',
```

JeffreyMFarley commented on src/PerCapita.jsx:
```diff
@@ -2,35 +2,58 @@ import './PerCapita.less'
+          <p>Map Shading</p>
+          <label className="u-visually-hidden" htmlFor="data-normalization">
+            Select the kind of shading to apply to the map
+          </label>
```
You should get rid of this since the drop-down disappeared.  It triggers a new WAVE warning

flacoman91 replied:
thanks for validating.  Will 🔪 this

sephcoster commented:
These changes look good. company_received (both) should also be on the trigger list. Thanks.
